components/expenses/ExpenseForm.js

```diff
@@ -123,6 +124,23 @@ const ExpenseFormBody = ({ formik, payoutProfiles, collective, autoFocusTitle, o
     }
   }, [payoutProfiles]);
 
+  // Load values from localstorage
+  React.useEffect(() => {
+    if (loggedInAccountId) {
+      formPersister?.setFormId(`expense-${collective.name}-${loggedInAccountId}`);
+    }
+
+    const formValues = formPersister?.loadValues();
+    if (formValues && isEqual(values, initialValues)) {
+      setValues(formValues);
+    }
```

Comment:
We can completely skip if there's no  loggedInAccountId or formPersister

```suggestion
    if (formPersister && loggedInAccountId) {
      formPersister.setFormId(`expense-${collective.name}-${loggedInAccountId}`);
      const formValues = formPersister.loadValues();
      if (formValues && isEqual(values, initialValues)) {
        setValues(formValues);
      }
    }
```